moved/src/block/root.rs

```diff
@@ -1,15 +1,20 @@
 use {
     crate::{
         primitives::{Address, Bytes, B2048, B256, U256},
-        types::transactions::ExtendedTxEnvelope,
+        types::{state::BlockResponse, transactions::ExtendedTxEnvelope},
     },
     alloy::{primitives::hex, rlp::RlpEncodable},
     std::fmt::Debug,
 };
 
-pub trait BlockRepository: Debug {
-    fn add(&mut self, block: ExtendedBlock);
-    fn by_hash(&self, hash: B256) -> Option<ExtendedBlock>;
+pub trait BlockQueries<Context>: Debug {
+    fn by_hash(&self, context: &Context, hash: B256) -> Option<BlockResponse>;
```

Comment:
Nit: I don't like the name `context` for this variable and type generic. In the implementation it's the backing store, which is just in memory for now, but presumably we'll have a persistent store in the future. I think a better name would reflect that this type represents the physical storage.

Reply:
I have a few suggestions such as:
* Storage
* PhysicalStorage
* BackingStore
* One of the above with suffix Access

I went with `Storage` for the generic and `storage` for the variable.

You also said you don't like the type generic. Did you mean you don't like the name of the generic parameter `Context` or something else?

Reply:
I'll add a few suggestions too:
- BlockStore
- BlockStorage

Reply:
I just meant the name `Context`. I like `Storage` fine.